lib/Drupal/service_container/Tests/LegacyMessengerTest.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\service_container\Tests\LegacyMessengerTest.
+ */
+
+namespace Drupal\service_container\Tests;
+
+use Drupal\service_container\Messenger\LegacyMessenger;
+
+class LegacyMessengerTest extends ServiceContainerIntegrationTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getInfo() {
+    return array(
+      'name' => 'LegacyMessenger',
+      'description' => 'Tests the messenger integration',
+      'group' => 'service_container',
+    );
+  }
+
+  public function testLegacyMessenger() {
+    /** @var \Drupal\service_container\Messenger\LegacyMessenger $messenger_service */
+
+    $messenger_service = $this->container->get('messenger');
+    $this->assertTrue($messenger_service instanceof LegacyMessenger);
+
+    $random_message = $this->randomString();
+    drupal_set_message($random_message, $messenger_service::WARNING);
+    $messages = $messenger_service->getMessages();
+    $warning_messages = $messages[$messenger_service::WARNING];
+    $this->assertTrue(in_array($random_message, $warning_messages));
```

Comment:
assertContains is better here: 

https://phpunit.de/manual/current/en/appendixes.assertions.html#appendixes.assertions.assertContains